components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -285,11 +307,15 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
   }
 
   public handleNameInput(event: KeyboardEvent): void {
-    if (!this.modifyID && !this.isNavigationKey(event)) {
+    if (!this.modifyID && !this.isNavigationKey(event) && !this.editingRule) {
       this.conflictError = false;
       this.ruleForm.controls.id.setValue(
         IdMapper.transform(this.ruleForm.controls.name.value.trim()));
     }
+
+    if (this.editingRule) {
+      this.ruleForm.markAsDirty();
+    }
```

Comment:
🤔 If you need to set dirty manually here, I think there is something broken in `chef-page`.

Reply:
I think the issue is that the conditions are actually an array of forms themselves. Making changes to the attributes doesn't appear to bubble up to the `ruleForm`. Not sure if `chef-page` is related